app/src/main/cpp/skyline/services/socket/sfdnsres/IResolver.cpp

```diff
@@ -2,7 +2,176 @@
+            std::memcpy(headerPtr, &header, headerSize);
+
+            if (header.addressLength == 0) {
+                std::memset(headerPtr + headerSize, 0, 4);
+            } else {
+                switch (current->ai_family) {
+                    case AF_INET: {
+                        struct SockAddrIn {
+                            u16 sin_family;
+                            u16 sin_port;
+                            u32 sin_addr;
+                            u8 sin_zero[8];
+                        };
+
+                        SockAddrIn serializedAddr{};
+                        const auto addr{*reinterpret_cast<sockaddr_in*>(current->ai_addr)};
+                        serializedAddr.sin_port = htons(addr.sin_port);
+                        serializedAddr.sin_family = htons(addr.sin_family);
+                        serializedAddr.sin_addr = htonl(addr.sin_addr.s_addr);
+                        std::memcpy(headerPtr + headerSize, &serializedAddr, sizeof(SockAddrIn));
+
+                        char addrStringBuf[64]{};
+                        inet_ntop(AF_INET, &addr.sin_addr, addrStringBuf, std::size(addrStringBuf));
+                        Logger::Info("Resolved host '{}' to IPv4 address {}", host, addrStringBuf);
+                        break;
+                    }
+                    case AF_INET6: {
+                        struct SockAddrIn6 {
+                            u16 sin6_family;
+                            u16 sin6_port;
+                            u32 sin6_flowinfo;
+                            u8 sin6_addr[16];
+                            u32 sin6_scope_id;
+                        };
+
+                        SockAddrIn6 serializedAddr{};
+                        const auto addr{*reinterpret_cast<sockaddr_in6*>(current->ai_addr)};
+                        serializedAddr.sin6_family = htons(addr.sin6_family);
+                        serializedAddr.sin6_port = htons(addr.sin6_port);
+                        serializedAddr.sin6_flowinfo = htonl(addr.sin6_flowinfo);
+                        serializedAddr.sin6_scope_id = htonl(addr.sin6_scope_id);
+                        std::memcpy(serializedAddr.sin6_addr, &addr.sin6_addr, sizeof(SockAddrIn6::sin6_addr));
+                        std::memcpy(headerPtr + headerSize, &serializedAddr, sizeof(SockAddrIn6));
+
+                        char addrStringBuf[64]{};
+                        inet_ntop(AF_INET6, &addr.sin6_addr, addrStringBuf, std::size(addrStringBuf));
+                        Logger::Info("Resolved host '{}' to IPv6 address {}", host, addrStringBuf);
+                        break;
+                    }
+                    default:
+                        std::memcpy(headerPtr + headerSize, current->ai_addr, addrSize);
+                        break;
+                }
+            }
+            if (current->ai_canonname) {
+                std::memcpy(headerPtr + addrSize, current->ai_canonname, canonnameSize);
```

Comment:
curOutputRegion.subspan(addrSize + headerSize).copy_from(span{current->ai_canonname, canonnameSize})

(is this right, rn it seems like it would overwrite the sock addr in)